### PR TITLE
Add Auth and Branding API

### PR DESCRIPTION
**Pull request by katerina-zaitsava**

### CODE REVIEW

File: src/api/auth/EmailVerificationApi.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    return this.apiClient.makeCall(`${this.path}`, 'POST', {
+      email,
+    }, {
+      headers: new Headers({
+        accountAccessToken,
+      }),
+    });
+  }
+
+  public verifyEmail(code: string, accountAccessToken: string): Promise<void>{
```
Comment (ilya-tishonok) [inline src/api/auth/EmailVerificationApi.ts:21]: add space after Promise<void>

File: src/api/auth/InvitesApi.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  [key in 'borrowerId' | 'intermediaryId']: string;
+};
+
+class InvitesApi {
+  protected path = '/invites'
+
+  constructor(
+    private apiClient: AuthApiClient,
+  ) {}
+
```
Comment (ilya-tishonok) [inline src/api/auth/InvitesApi.ts:17]: remove empty line

File: src/api/auth/ResetPasswordApi.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  public resetPassword(password: string, resetPasswordToken: string): Promise<void> {
+    return this.apiClient.makeCall(`${this.path}/${resetPasswordToken}`, 'POST', {
+      password,
+    });
+  };
+
+  public getResetPasswordTokenInfo(resetPasswordToken: string): Promise<GetResetPasswordTokenInfoResponseParams> {
+    return this.apiClient.makeCall(`${this.path}/${resetPasswordToken}`, 'GET');
+  };
```
[the remark this answers is not in the document]
Comment (ilya-tishonok) [reply src/api/auth/ResetPasswordApi.ts:30]: pls fix in all places

**Review by ilya-tishonok — APPROVED** [review]

File: src/api/auth/ResetPasswordApi.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (katerina-zaitsava) [reply src/api/auth/ResetPasswordApi.ts]: idk, my IDE didn't mind so I put ; everywhere :D
removed now

File: src/api/auth/InvitesApi.ts (hunk 2 above)
Comment (katerina-zaitsava) [reply src/api/auth/InvitesApi.ts:17]: ✔️

File: src/api/auth/EmailVerificationApi.ts (hunk 1 above)
Comment (katerina-zaitsava) [reply src/api/auth/EmailVerificationApi.ts]: ✔️

File: src/AuthApiClient.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { FetchOptions, HTTP_METHOD, RequestBody } from './ApiClient';
+
+class AuthApiClient extends AuthorizedApiClient {
+  protected entity;
+
+  constructor(
+    baseUrl: string,
+    clientId: string,
+    clientSecret: string,
+    entity: 'borrowers' | 'intermediaries',
```
Comment (alexey-maslovsky) [inline src/AuthApiClient.ts:11]: you can do like "protected entity: 'borrowers' | 'intermediaries'," and you don't need "protected entity"/"this.entity = entity;"

File: src/api/auth/InvitesApi.ts (hunk 2 above)
Comment (alexey-maslovsky) [reply src/api/auth/InvitesApi.ts:16]: ilya-eslint-tishonok

File: src/AuthApiClient.ts (hunk 4 above)
Comment (katerina-zaitsava) [reply src/AuthApiClient.ts]: ✔️

**Pull request merged**